embulk-standards/src/main/java/org/embulk/standards/JsonParserPlugin.java

```diff
@@ -99,12 +131,79 @@ private PageBuilder newPageBuilder(Schema schema, PageOutput output)
+            case UNESCAPE:
+                Iterable<CharSource> lines = Lists.transform(CharStreams.readLines(new BufferedReader(new InputStreamReader(in))),
+                        invalidEscapeStringFunction(policy));
+                return new JsonParser().open(new ReaderInputStream(CharSource.concat(lines).openStream()));
+            case PASSTHROUGH:
+            default:
+                return new JsonParser().open(in);
+        }
+    }
+
+    Function<String, CharSource> invalidEscapeStringFunction(final InvalidEscapeStringPolicy policy) {
+        return new Function<String, CharSource>() {
+            final Pattern p = Pattern.compile("\\p{XDigit}+");
+            @Override
+            public CharSource apply(@Nullable String input) {
+                assert input != null;
+                if (policy == InvalidEscapeStringPolicy.PASSTHROUGH) {
+                    return CharSource.wrap(input);
+                }
+                StringBuilder s = new StringBuilder();
+                char[] charArray = input.toCharArray();
+                for (int i=0 ; i < charArray.length ; i++) {
+                    char c = charArray[i];
+                    if (c == '\\') {
+                        if (charArray.length > i + 1) {
+                            char next = charArray[i + 1];
+                            switch (next) {
+                                case 'b':
+                                case 'f':
+                                case 'n':
+                                case 'r':
+                                case 't':
+                                case '"':
+                                case '\\':
+                                case '/':
+                                    s.append(c);
+                                    break;
+                                case 'u': // hexstring such as \u0001
+                                    if (charArray.length > i + 5) {
+                                        char[] hexChars = {charArray[i + 2], charArray[i + 3], charArray[i + 4], charArray[i + 5]};
+                                        String hexString = new String(hexChars);
+                                        if (p.matcher(hexString).matches()) {
+                                            s.append(c);
+                                        }
+                                    }
+                                    break;
+                                default:
+                                    switch (policy) {
+                                        case SKIP:
+                                            i++;
+                                            break;
+                                        case UNESCAPE:
+                                            break;
+                                    }
+                                    break;
+                            }
+                        }
+                    } else {
+                        s.append(c);
```

Comment:
No, if `\\` has at the end of line , it will removed at SKIP or UNESCAPE. I will check test and fix it.

```
{"a":"b}
\\
```

convert to:

```
{"a":"b}
```